test/e2e/app/app.go

```diff
@@ -439,18 +446,28 @@ func (app *Application) ExtendVote(_ context.Context, req *abci.RequestExtendVot
 		panic(fmt.Errorf("received call to ExtendVote at height %d, when vote extensions are disabled", appHeight))
 	}
 
-	ext := make([]byte, binary.MaxVarintLen64)
-	// We don't care that these values are generated by a weak random number
-	// generator. It's just for test purposes.
-	//nolint:gosec // G404: Use of weak random number generator
-	num := rand.Int63n(voteExtensionMaxVal)
-	extLen := binary.PutVarint(ext, num)
-
 	if app.cfg.VoteExtensionDelay != 0 {
 		time.Sleep(app.cfg.VoteExtensionDelay)
 	}
 
-	app.logger.Info("generated vote extension", "num", num, "ext", fmt.Sprintf("%x", ext[:extLen]), "height", appHeight)
+	var ext []byte
+	var extLen int
+	if app.cfg.VoteExtensionSize != 0 {
+		ext = make([]byte, app.cfg.VoteExtensionSize)
+		if _, err := rand.Read(ext); err != nil {
+			panic(fmt.Errorf("could not extend vote. Len:%d", len(ext)))
+		}
+		extLen = len(ext)
+	} else {
+		ext = make([]byte, 8)
+		if num, err := rand.Int(rand.Reader, big.NewInt(voteExtensionMaxVal)); err != nil {
+			panic(fmt.Errorf("could not extend vote. Len:%d", len(ext)))
+		} else {
+			extLen = binary.PutVarint(ext, num.Int64())
```

Comment:
If you're converting to int64, do you really need the `big` library in line 463?

More generally, wouldn't it be simpler to keep the old code inside this "else" (as you did in `VerifyVoteExtension`?

Reply:
The change was because I updated the code to use `crypto/rand`, which returns a `big.Int`.
It was laborious way of getting rid of a lint :)